### PR TITLE
4.3 Prep

### LINKED ISSUE #575: [RELIABILITY/UX] Make yt-dlp the default for new installations
I think it's fair to say that is has become the best fork yet and fixes a lot of yt-dl's and yt-dlc's shortcomings.

### LINKED ISSUE #600: Error when selecting another "Max quality"
Hi,

When on any subscription I choose another "Max quality" than "Best" looks like it is not building correctly the youtube-dl arguments (missing "]", puting just '(mp4)[height=720'):

`2022-05-04T15:40:43.784Z INFO: YoutubeDL-Material v4.2 started on PORT 17442
2022-05-04T15:40:43.791Z VERBOSE: Watching test with delay interval of 30000000
2022-05-04T15:40:43.802Z VERBOSE: Subscription: getting videos for subscription test with args: --dump-json,-o,"subscriptions/playlists/test/%(id)s.%(ext)s",-ciw,--write-info-json,--print-json,-f,bestvideo[height<=720]+bestaudio/best[height<=720],--merge-output-format,mp4,--download-archive,,--write-thumbnail
2022-05-04T15:40:55.216Z VERBOSE: Subscription: finished check for test
2022-05-04T15:40:55.278Z DEBUG: Inserted doc into download_queue
2022-05-04T15:40:55.284Z DEBUG: Inserted doc into download_queue
2022-05-04T15:40:55.290Z DEBUG: Inserted doc into download_queue
2022-05-04T15:40:55.801Z VERBOSE: Collecting info for download 912d35e7-1f19-4075-abf8-fed9c4a95509
2022-05-04T15:40:55.805Z VERBOSE: Collecting info for download 2d4d49df-fb16-4f98-8ec8-1a1744257a21
2022-05-04T15:40:55.809Z VERBOSE: Collecting info for download 6514bb69-8d62-4255-93d2-aa4129dc41c1
2022-05-04T15:40:55.852Z VERBOSE: youtube-dl args being used: -o,subscriptions/playlists/test/%(id)s.%(ext)s,--write-info-json,--print-json,-f,'(mp4)[height=720',--download-archive,/app/subscriptions/archives/test/merged_video.txt,--write-thumbnail
2022-05-04T15:40:55.871Z VERBOSE: youtube-dl args being used: -o,subscriptions/playlists/test/%(id)s.%(ext)s,--write-info-json,--print-json,-f,'(mp4)[height=720',--download-archive,/app/subscriptions/archives/test/merged_video.txt,--write-thumbnail
2022-05-04T15:40:55.890Z VERBOSE: youtube-dl args being used: -o,subscriptions/playlists/test/%(id)s.%(ext)s,--write-info-json,--print-json,-f,'(mp4)[height=720',--download-archive,/app/subscriptions/archives/test/merged_video.txt,--write-thumbnail
2022-05-04T15:41:12.969Z ERROR: Error while retrieving info on video with URL https://www.youtube.com/watch?v=DkPpu5bRgAQ with the following message: Error: Command failed with exit code 1: /app/node_modules/youtube-dl/bin/youtube-dl -o subscriptions/playlists/test/%(id)s.%(ext)s --write-info-json --print-json -f '(mp4)[height=720' --write-thumbnail --dump-json http://www.youtube.com/watch?v=DkPpu5bRgAQ
ERROR: requested format not available
2022-05-04T15:41:13.378Z ERROR: Error while retrieving info on video with URL https://www.youtube.com/watch?v=9DXMDzqA-UI with the following message: Error: Command failed with exit code 1: /app/node_modules/youtube-dl/bin/youtube-dl -o subscriptions/playlists/test/%(id)s.%(ext)s --write-info-json --print-json -f '(mp4)[height=720' --write-thumbnail --dump-json http://www.youtube.com/watch?v=9DXMDzqA-UI
ERROR: requested format not available
2022-05-04T15:41:15.152Z ERROR: Error while retrieving info on video with URL https://www.youtube.com/watch?v=zA52uNzx7Y4 with the following message: Error: Command failed with exit code 1: /app/node_modules/youtube-dl/bin/youtube-dl -o subscriptions/playlists/test/%(id)s.%(ext)s --write-info-json --print-json -f '(mp4)[height=720' --write-thumbnail --dump-json http://www.youtube.com/watch?v=zA52uNzx7Y4
ERROR: requested format not available
2022-05-04T15:41:15.872Z VERBOSE: Disabling checking downloads as none are available.`

### LINKED ISSUE #662: [FEATURE] Using npm start
**Is your feature request related to a problem? Please describe.**
By default the youtubedl-m use PM2 runtime, that need installed manually `npm install -g pm2` that need use root permission. By using `npm start` in package.json. We can include pm2 in backend dependencies and set npm start to `pm2-runtime --raw pm2.config.js` then dropping the need install pm2-runtime with root permission.

Looking from https://github.com/Tzahi12345/YoutubeDL-Material/blob/6fa4296edfbc7d3201a9722558e171d29f486537/backend/package.json#L8 seems youtubedl-m used to be using `nodemon` and `spoiler: i tried and it not working`.

**Describe the solution you'd like**
1. Change `npm start` to `pm2-runtime --raw pm2.config.js`
2. Add pm2-runtime to backend dependencies

**Additional context**
Using `npm start` will likely easier reproducible runtime, as user doesn't need to manually install pm2.

https://github.com/Tzahi12345/YoutubeDL-Material/pull/657

### PR DESCRIPTION
**Pull request by Tzahi12345**
Summary:
* Updated UI for creating/modifying playlists (see demo below)
* Playlists are now file type agnostic (that is, you can now mix and match audio files and video files into a playlist)
* Subscription file cards now use the new ones on the home page
* Fixed Twitch chat downloads
  * They are much faster now! Downside is a python dependency is now required ([Twitch-Chat-Downloader](https://github.com/PetterKraabol/Twitch-Chat-Downloader)) as the new Twitch API does not support downloading chat
  * A client secret must now be supplied in addition to the client ID
* Fixed issue where subscription downloads could not set quality other than max (fixes #600)
* Changed default downloader from youtube-dl to yt-dlp (fixes #575)
* Version updated to v4.3
* Updated translations
* Updated API/models
* Various code cleanup
* Fixed issue where playlist order could not be rearranged
* Removed recommendations to use `nightly`
* pm2 moved from global install to local (closes #662)

**Demo**:
https://user-images.githubusercontent.com/707413/174672249-00990e94-2886-4748-b49a-e2f3761d80c1.mp4

### Done testing! Once this is merged, v4.3 will be released!

### CODE REVIEW

**Comment by GlassedSilver** [comment]
Maybe removing the recommendations to use the nightly branch would be a good idea.

Same for docker-compose.yml.

File: backend/twitch.js [hunk 1]
```diff
@@ -1,90 +1,53 @@
...
+
+const moment = require('moment');
+const fs = require('fs-extra')
+const path = require('path');
+
+async function getCommentsForVOD(clientID, clientSecret, vodId) {
+    const { promisify } = require('util');
+    const child_process = require('child_process');
+    const exec = promisify(child_process.exec);
+    const result = await exec(`tcd --video ${vodId} --client-id ${clientID} --client-secret ${clientSecret} --format json -o appdata`, {stdio:[0,1,2]});
```
[the remark this answers is not in the document]
Comment (Tzahi12345) [reply backend/twitch.js:12]: Not sure how to address this... We can ignore it or figure out our own way to sanitize.

**Comment by martadinata666** [comment]
Add to wiki solution for RPI4 or other devices that mongodb throw exception.
https://github.com/Tzahi12345/YoutubeDL-Material/issues/614#issuecomment-1142216713

and add some references
* https://jira.mongodb.org/browse/DOCS-14684
* https://github.com/docker-library/mongo/issues/485
* https://www.mongodb.com/community/forums/t/mongodb-5-0-cpu-intel-g4650-compatibility/116610
* https://www.mongodb.com/community/forums/t/core-dump-on-mongodb-5-0-on-rpi-4/115291

**Comment by Tzahi12345** [comment]
> Add to wiki solution for RPI4 or other devices that mongodb throw exception (132).

I'm honestly down to just target mongo:4.4.6

Not sure if there's a reason to be at v5 anyways... @GlassedSilver thoughts?

**Comment by martadinata666** [comment]
> > Add to wiki solution for RPI4 or other devices that mongodb throw exception (132).
> 
> I'm honestly down to just target mongo:4.4.6
> 
> Not sure if there's a reason to be at v5 anyways... @GlassedSilver thoughts?

Yeah, updating compose to target `mongo:4.4` also will do, it just we also need to add reason `why` because as you know some folks just `why not :latest?`

**Comment by Tzahi12345** [comment]
> Yeah, updating compose to target `mongo:4.4` also will do, it just we also need to add reason `why` because as you know some folks just `why not :latest?`

I did just try on a test instance downgrading and it was not pretty. I'll probably go with your first suggestion and add something to the README for rpi 4

**Review by GlassedSilver — CHANGES_REQUESTED** [review]
small change I would like to ask for from what I was able to spot so far, other than that LGTM :)

File: src/app/components/unified-file-card/unified-file-card.component.html [hunk 2]
```diff
@@ -34,10 +34,10 @@
                 <mat-icon>restore</mat-icon><ng-container i18n="Delete and redownload subscription video button">Delete and redownload</ng-container>
             </button>
             <button *ngIf="file_obj.sub_id && use_youtubedl_archive" (click)="emitDeleteFile(true)" mat-menu-item>
-                <mat-icon>delete_forever</mat-icon><ng-container i18n="Delete forever subscription video button">Delete forever</ng-container>
+                <mat-icon>delete_forever</mat-icon><ng-container i18n="Delete forever subscription video button">Delete and don't download again</ng-container>
```
Comment (GlassedSilver) [inline src/app/components/unified-file-card/unified-file-card.component.html:37]: The original wording you used in this PR seemed a bit technical. A better UX wording would be to refer to this as "Delete and don't download again" like we discussed on Discord, so I felt free to commit this change myself. :) Same as below

File: src/app/components/unified-file-card/unified-file-card.component.html [hunk 3]
```diff
@@ -34,10 +34,10 @@
...
             </button>
             <button *ngIf="!file_obj.sub_id" (click)="emitDeleteFile()" mat-menu-item><mat-icon>delete</mat-icon><ng-container i18n="Delete video button">Delete</ng-container></button>
-            <button *ngIf="!file_obj.sub_id && use_youtubedl_archive" (click)="emitDeleteFile(true)" mat-menu-item><mat-icon>delete_forever</mat-icon><ng-container i18n="Delete and blacklist video button">Delete and blacklist</ng-container></button>
+            <button *ngIf="!file_obj.sub_id && use_youtubedl_archive" (click)="emitDeleteFile(true)" mat-menu-item><mat-icon>delete_forever</mat-icon><ng-container i18n="Delete and don't download again">Delete and don't download again</ng-container></button>
```
Comment (GlassedSilver) [inline src/app/components/unified-file-card/unified-file-card.component.html:40]: Refer to my comment for above (line 37)

**Review by GlassedSilver — APPROVED** [review]
LGTM :)

**Review by GlassedSilver — APPROVED** [review]

**Pull request merged**